conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegration.java

```diff
@@ -12,88 +12,61 @@
 
     // Wrapper to extract bitmovinPlayer helper methods
     private BitmovinPlayerHelper playerHelper;
 
     // Helper
     private Boolean adStarted = false;
+    private Boolean isSessionActive = false;
 
-    public ConvivaAnalytics(Player player, String customerKey, Context context) {
+    public ConvivaAnalyticsIntegration(Player player, String customerKey, Context context) {
         this(player, customerKey, context, new ConvivaConfig());
     }
 
-    public ConvivaAnalytics(Player player,
-                            String customerKey,
-                            Context context,
-                            ConvivaConfig config) {
-        this(player, customerKey, context, config, null);
-    }
-
-    public ConvivaAnalytics(Player player,
-                            String customerKey,
-                            Context context,
-                            ConvivaConfig config,
-                            Client client) {
+    public ConvivaAnalyticsIntegration(Player player,
+                                       String customerKey,
+                                       Context context,
+                                       ConvivaConfig config) {
         this.bitmovinPlayer = player;
         this.playerHelper = new BitmovinPlayerHelper(player);
         this.config = config;
 
-        SystemInterface androidSystemInterface = AndroidSystemInterfaceFactory.buildSecure(context);
-        if (androidSystemInterface.isInitialized()) {
-            SystemSettings systemSettings = new SystemSettings();
-            systemSettings.allowUncaughtExceptions = false;
-
-            if (config.isDebugLoggingEnabled()) {
-                systemSettings.logLevel = SystemSettings.LogLevel.DEBUG;
-            }
-
-            SystemFactory androidSystemFactory = new SystemFactory(androidSystemInterface, systemSettings);
-            ClientSettings clientSettings = new ClientSettings(customerKey);
-
+        if(BuildConfig.DEBUG) {
+            Map<String, Object> settings = new HashMap<String, Object>();
             if (config.getGatewayUrl() != null) {
-                clientSettings.gatewayUrl = config.getGatewayUrl();
+                settings.put(ConvivaSdkConstants.GATEWAY_URL, config.getGatewayUrl());
             }
-
-            if (client != null) {
-                this.client = client;
-            } else {
-                this.client = new Client(clientSettings, androidSystemFactory);
+            if (config.isDebugLoggingEnabled()) {
+                settings.put(ConvivaSdkConstants.LOG_LEVEL, ConvivaSdkConstants.LogLevel.DEBUG);
             }
-
-            attachBitmovinEventListeners();
+            ConvivaAnalytics.init(context, customerKey, settings);
+        } else {
+            ConvivaAnalytics.init(context, customerKey);
         }
-    }
 
-    private void ensureConvivaSessionIsCreatedAndInitialized() {
-        if (!isSessionActive()) {
-            internalInitializeSession();
```

Comment:
Hmm. It's a good point to consider. Although sessions are now handled internally in Conviva, I did see that having non-matching - that is, an unequal amount of - calls to `convivaVideoAnalytics.reportPlaybackRequested(contentMetadataBuilder.build());` and `convivaVideoAnalytics.reportPlaybackEnded();` results in Touchstone errors. This is why I reinstated a simpler form of session tracking, simply to track these methods and ensure they are "balanced". 

Having said that, you are quite right and I'm thinking that we should also reinstate that method so that we don't require the explicit `initializeSession()`. So I'll get on that!